neurolib/models/multimodel/builder/fitzhugh_nagumo.py

```diff
@@ -0,0 +1,179 @@
+        """
+        :param params: parameters of the FitzHugh-Nagumo mass
+        :type params: dict|None
+        :param seed: seed for random number generator
+        :type seed: int|None
+        """
+        fhn_mass = FitzHughNagumoMass(params, seed=seed)
+        fhn_mass.index = 0
+        super().__init__(neural_masses=[fhn_mass])
+
+    def _sync(self):
+        return []
+
+
+class FitzHughNagumoNetwork(Network):
+    """
+    Whole brain network of FitzHugh-Nagumo oscillators.
+    """
+
+    name = "FitzHugh-Nagumo network"
+    label = "FHNnet"
+
+    sync_variables = ["network_x", "network_y"]
+
+    def __init__(
+        self, connectivity_matrix, delay_matrix, mass_params=None, x_coupling="diffusive", y_coupling="none", seed=None,
+    ):
+        """
+        :param connectivity_matrix: connectivity matrix for between nodes
+            coupling, typically DTI structural connectivity, matrix as [to,
+            from]
+        :type connectivity_matrix: np.ndarray
+        :param delay_matrix: delay matrix between nodes, typically derived from
+            length matrix, if None, delays are all zeros, in ms, matrix as
+            [to, from]
+        :type delay_matrix: np.ndarray|None
+        :param mass_params: parameters for each Hopf normal form neural
+            mass, if None, will use default
+        :type mass_params: list[dict]|dict|None
+        :param x_coupling: how to couple `x` variables in the nodes,
+            "diffusive", "additive", or "none"
+        :type x_coupling: str
+        :param y_coupling: how to couple `y` variables in the nodes,
+            "diffusive", "additive", or "none"
+        :type y_coupling: str
+        :param seed: seed for random number generator
+        :type seed: int|None
+        """
+        mass_params = self._prepare_mass_params(mass_params, connectivity_matrix.shape[0])
+        seeds = self._prepare_mass_params(seed, connectivity_matrix.shape[0], native_type=int)
+
+        nodes = []
+        for i, node_params in enumerate(mass_params):
+            node = FitzHughNagumoNetworkNode(params=node_params, seed=seeds[i])
+            node.index = i
+            node.idx_state_var = i * node.num_state_variables
+            nodes.append(node)
+
+        super().__init__(
+            nodes=nodes, connectivity_matrix=connectivity_matrix, delay_matrix=delay_matrix,
+        )
+        # get all coupling variables
+        all_couplings = [mass.coupling_variables for node in self.nodes for mass in node.masses]
+        # assert they are the same
+        assert all(all_couplings[0] == coupling for coupling in all_couplings)
+        # invert as to name: idx
+        self.coupling_symbols = {v: k for k, v in all_couplings[0].items()}
+
+        self.x_coupling = x_coupling
+        self.y_coupling = y_coupling
+
+    def _sync(self):
+        return self._couple(self.x_coupling, "x") + self._couple(self.y_coupling, "y") + super()._sync()
```

Comment:
Couldn't this be automated in the superclass actually? 
```python
return np.sum([<look upver couplings here>], super().sync())
```

Reply:
I moved this part to `Network` base class and create a new way of how couplings are determined for the basic models